test/jdk/javax/swing/JCheckBox/4449413/bug4449413.java

```diff
@@ -25,87 +25,141 @@
  * @bug 4449413
  * @summary Tests that checkbox and radiobuttons' check marks are visible when background is black
  * @author Ilya Boyandin
- * @library /test/lib
- * @modules java.desktop/sun.awt
- * @build jdk.test.lib.Platform
- * @run applet/manual=yesno bug4449413.html
+ * @run main/manual bug4449413
  */
 
-import javax.swing.*;
-import javax.swing.plaf.metal.*;
-import java.awt.event.*;
-import java.awt.*;
+import javax.swing.AbstractButton;
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JCheckBoxMenuItem;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JRadioButton;
+import javax.swing.JRadioButtonMenuItem;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.plaf.metal.DefaultMetalTheme;
+import javax.swing.plaf.metal.MetalLookAndFeel;
+import javax.swing.plaf.metal.MetalTheme;
+import javax.swing.plaf.metal.OceanTheme;
+import java.awt.Color;
+import java.awt.GridLayout;
+import java.awt.Insets;
+import java.awt.event.ActionListener;
+import java.awt.event.ItemEvent;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class bug4449413 extends JFrame {
+
+    private static final String INSTRUCTIONS = """
+            This test is for MetalLookAndFeel only.
+
+            There are eight controls with black backgrounds.
+            Four enabled (on the left side) and four disabled (on the right side)
+            checkboxes and radiobuttons.
+
+            1. If at least one of the controls' check marks is not visible:
+               the test fails.
+
+            2. Uncheck the "Use Ocean Theme" check box.
+               If now at least one of the controls' check marks is not visible:
+               the test fails.
+            """;
+
+    private static final CountDownLatch latch = new CountDownLatch(1);
+    private static volatile boolean failed = true;
+
+    private final MetalTheme defaultMetalTheme = new DefaultMetalTheme();
+    private final MetalTheme oceanTheme = new OceanTheme();
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+
+        SwingUtilities.invokeLater(() -> new bug4449413().createAndShowGUI());
+
+        boolean timeoutHappened = !latch.await(2, TimeUnit.MINUTES);
+
+        if (timeoutHappened || failed) {
+            throw new RuntimeException("Test failed!");
+        }
```

Comment:
Shall the test dispose of the frame if it times out?

Reply:
Actually `jtreg` handles it. But I've updated the test to support standalone mode.